pkg/ami/ami_test.go

```diff
@@ -0,0 +1,112 @@
+package ami_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/stretchr/testify/mock"
+	"github.com/weaveworks/eksctl/pkg/ami"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/testutils/mockprovider"
+)
+
+func TestUseAMI(t *testing.T) {
+	amiTests := []struct {
+		blockDeviceMappings []*ec2.BlockDeviceMapping
+		rootDeviceName      string
+		description         string
+	}{
+		{
+			description:    "Root device mapping not at index 0 (Windows AMIs in some regions)",
+			rootDeviceName: "/dev/sda1",
+			blockDeviceMappings: []*ec2.BlockDeviceMapping{
+				{
+					DeviceName:  aws.String("xvdca"),
+					VirtualName: aws.String("ephemeral0"),
+				},
+				{
+					DeviceName:  aws.String("xvdcb"),
+					VirtualName: aws.String("ephemeral1"),
+				},
+				{
+					DeviceName: aws.String("/dev/sda1"),
+					Ebs: &ec2.EbsBlockDevice{
+						Encrypted: aws.Bool(true),
+					},
+				},
+			},
+		},
+		{
+			description:    "Only one device mapping (AL2 AMIs)",
+			rootDeviceName: "/dev/sda1",
+			blockDeviceMappings: []*ec2.BlockDeviceMapping{
+				{
+					DeviceName: aws.String("/dev/sda1"),
+					Ebs: &ec2.EbsBlockDevice{
+						Encrypted: aws.Bool(true),
+					},
+				},
+			},
+		},
+		{
+			description:    "Different root device name",
+			rootDeviceName: "/dev/xvda",
+			blockDeviceMappings: []*ec2.BlockDeviceMapping{
+				{
+					DeviceName:  aws.String("xvdca"),
+					VirtualName: aws.String("ephemeral0"),
+				},
+				{
+					DeviceName: aws.String("/dev/xvda"),
+					Ebs: &ec2.EbsBlockDevice{
+						Encrypted: aws.Bool(true),
+					},
+				},
+				{
+					DeviceName:  aws.String("xvdcb"),
+					VirtualName: aws.String("ephemeral1"),
+				},
+			},
+		},
+	}
+
+	for i, tt := range amiTests {
+		t.Run(fmt.Sprintf("%d: %s", i, tt.description), func(t *testing.T) {
+			mockProvider := mockDescribeImages(tt.blockDeviceMappings, tt.rootDeviceName)
+			err := ami.Use(mockProvider.MockEC2(), &api.NodeGroup{
```

Comment:
Shouldn't we check that the actual ng gets the `VolumeEncrypted` correctly set?

Reply:
Good point, I'll add the check.